src/lib.rs

```diff
@@ -328,18 +433,26 @@ fn calculate_waste(
     accumulated_weight: u32,
     estimated_fee: u64,
 ) -> u64 {
-    let mut waste: u64 = 0;
+    // waste =  weight*(target feerate - long term fee rate) + cost of change + excess
+    // weight - total weight of selected inputs
+    // cost of change - includes the fees paid on this transaction's change output plus the fees that will need to be paid to spend it later. If there is no change output, the cost is 0.
+    // excess - refers to the difference between the sum of selected inputs and the amount we need to pay (the sum of output values and fees). There shouldn’t be any excess if there is a change output.
 
-    if let Some(long_term_feerate) = options.long_term_feerate {
-        waste += (estimated_fee as f32
-            - selected_inputs.len() as f32 * long_term_feerate * accumulated_weight as f32)
-            .ceil() as u64;
-    }
-
-    if options.excess_strategy != ExcessStrategy::ToDrain {
-        waste += accumulated_value - options.target_value - estimated_fee;
-    } else {
-        waste += options.drain_cost;
+    let mut waste: u64 = 0;
+    if options.excess_strategy == ExcessStrategy::ToDrain {
```

Comment:
Use the same if else structure which was there originally.
You have missed some cases and if else conditions are repetitive.